src/preferences/upgrade.cpp

```diff
@@ -29,6 +30,45 @@ Upgrade::Upgrade()
 Upgrade::~Upgrade() {
 }
 
+namespace {
+// mapping to proactively move users to the new all-shader waveform types
+WaveformWidgetType::Type upgradeToAllShaders(WaveformWidgetType::Type waveformType) {
+    // TODO: convert `WaveformWidgetType::Type` to an enum class then shorten more `using enum ...`
+    using WWT = WaveformWidgetType;
+    switch (waveformType) {
+    case WWT::EmptyWaveform:
+    case WWT::SoftwareSimpleWaveform:
+    case WWT::SoftwareWaveform:
+    case WWT::GLVSyncTest:
+    case WWT::QtVSyncTest:
+    case WWT::AllShaderRGBWaveform:
+    case WWT::AllShaderLRRGBWaveform:
+    case WWT::AllShaderFilteredWaveform:
+    case WWT::AllShaderSimpleWaveform:
+    case WWT::AllShaderHSVWaveform:
+    case WWT::Count_WaveformwidgetType:
+        return waveformType;
+    case WWT::QtSimpleWaveform:
+    case WWT::GLSimpleWaveform:
+        return WaveformWidgetType::AllShaderSimpleWaveform;
+    case WWT::GLFilteredWaveform:
+    case WWT::GLSLFilteredWaveform:
+        return WaveformWidgetType::AllShaderFilteredWaveform;
+    case WWT::QtWaveform:
+    case WWT::HSVWaveform:
+    case WWT::RGBWaveform:
+    case WWT::GLRGBWaveform:
+    case WWT::GLSLRGBWaveform:
+    case WWT::QtRGBWaveform:
+    case WWT::GLSLRGBStackedWaveform:
+        return WaveformWidgetType::AllShaderHSVWaveform;
+    case WWT::QtHSVWaveform:
```

Comment:
sorry, I was a little sloppy with my previous suggestion. I double checked and now it should be correct. I also forgot to apply the `WWT` alias, so do these for the others as well. Please amend and force push. Thank you.
```suggestion
    case WWT::QtWaveform:
    case WWT::RGBWaveform:
    case WWT::GLRGBWaveform:
    case WWT::GLSLRGBWaveform:
    case WWT::QtRGBWaveform:
    case WWT::GLSLRGBStackedWaveform:
        return WWT::AllShaderRGBWaveform;
    case WWT::HSVWaveform:
    case WWT::QtHSVWaveform:
```

Reply:
I will make the suggestion changes this weekend and with that hopefully this can be merged.

Reply:
I'm sorry. This could've been merged a week ago if I didn't add the previous suggestion which had introduced a bug. My fault.